app/jobs/google/search_keyword_job.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Google
+  class ClientServiceError < StandardError; end
+
+  class SearchKeywordJob < ApplicationJob
+    queue_as :default
+
+    retry_on ClientServiceError, ArgumentError
+
+    def perform(keyword)
+      html_result = ClientService.new(keyword: keyword.name).call
+
+      raise ClientServiceError unless html_result
+
+      update_keyword keyword.id, ParserService.new(html_response: html_result).call
+    rescue ClientServiceError, ArgumentError => e
+      job_failed keyword.id
+
+      raise e
```

Comment:
@ankitkalia1195  @junan 
I would love to hear about your advises for this: I want to update status `failed` BUT also to enable the retry. 
Is it acceptable to `rescue` an error and to raise it inside the rescue block? Or do you see another way to perform that?

Many thanks! :) 

PS: This PR is still in draft, to be reviewed by myself, no need to review everything at the moment ;-)

Reply:
@malparty This seems to to be the only way, but maybe you can remove `e` and write just `raise`, it will re-raise last expcetion.

Reply:
Will do! :)

Reply:
@ankitkalia1195 ✅ Done. 
Though it was harder than expected: I needed to find the proper way to authenticate `request` tests (for web controller, not api) with Devise helpers. The `sign_in` method exposed by Devise helpers was hidden by my previous helpers~ 

But after all, I believe it's good -- and probably cleaner. 

Cf commit @malparty
[.#38 Add request tests to assert job being enqueued](https://github.com/malparty/google-search-ruby/pull/68/commits/502cfcdba2ee15511c7e165ee68cdd31214589f0)